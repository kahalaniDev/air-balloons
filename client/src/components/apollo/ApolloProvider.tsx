import React from "react";
import { MockedProvider } from "@apollo/client/testing";
import { ApolloProvider as ApolloNativeProvider } from "@apollo/client";
import { apolloClient } from "../../infrastructure/apollo/client";
import { loginGraphqlMock } from "../../features/account/api/login/mock/loginGraphqlMock";
import { getBalloonsGraphqlMock } from "../../features/balloons/api/getBalloons/mock/getBalloonsGraphqlMock";
import { getBalloonGraphqlMock } from "../../features/balloons/api/getBalloon/mock/getBalloonGraphqlMock";
import { addBalloonGraphqlMock } from "../../features/balloons/api/addBalloon/mock/addBalloonGraphqlMock";

interface Props {
  useMocks?: boolean;
}

const ApolloProvider: React.FC<Props> = ({ useMocks, children }) => {
  if (useMocks)
    return (
      <MockedProvider
        mocks={[
          ...loginGraphqlMock,
          ...getBalloonsGraphqlMock,
          ...getBalloonGraphqlMock,
          ...addBalloonGraphqlMock,
        ]}
      >
        <>{children}</>
      </MockedProvider>
    );
  return (
    <ApolloNativeProvider client={apolloClient}>
      <>{children}</>
    </ApolloNativeProvider>
  );
};

export default ApolloProvider;
