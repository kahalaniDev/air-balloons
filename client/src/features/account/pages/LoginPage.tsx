import React from "react";
import LoginForm from "../components/LoginForm";
import { Avatar, Typography } from "@mui/material";
import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
import { LoginLayout } from "../styles/accountStyles";

type Props = {};

const LoginPage: React.FC<Props> = () => {
  return (
    <LoginLayout elevation={4}>
      <Avatar sx={{ m: 1, bgcolor: "secondary.main" }}>
        <LockOutlinedIcon />
      </Avatar>
      <Typography component="h1" variant="h5">
        Sign in
      </Typography>
      <LoginForm />
    </LoginLayout>
  );
};

export default LoginPage;
