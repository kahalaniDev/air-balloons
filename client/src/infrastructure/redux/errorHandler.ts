import { ApolloError, isApolloError } from "@apollo/client";
import axios, { AxiosError } from "axios";
import { GraphQLError } from "graphql";
import { IAppError } from "../../models/interfaces";

const GLOBAL_STATUS_CODE = 500;
const INTERNAL_ERR_MSG = "Internal server problem, please try again";
const NEWTWORK_ERR_MSG = "Unable to access the server, please try again";

export const errorHandler = (err: unknown) => {
  if (axios.isAxiosError(err)) return axiosErrorBuilder(err);
  if (err instanceof Error && isApolloError(err))
    return apolloErrorBuilder(err);
  return {
    statusCode: 500,
    message: INTERNAL_ERR_MSG,
  };
};

const axiosErrorBuilder = (axiosError: AxiosError) => {
  console.log(axiosError);
  const statusText = axiosError.response?.statusText
    ? axiosError.response!.statusText
    : "";
  const statusCode = axiosError.response!.status;
  switch (statusText) {
    case "Not Found":
      return { statusCode, message: axiosError.response!.data };
    case "Bad Request":
      return { statusCode, message: axiosError.response!.data };
    case "Conflict":
      return { statusCode, message: axiosError.response!.data };
    case "Unauthorized":
      return { statusCode, message: axiosError.response!.data };
    default:
      return { statusCode, message: INTERNAL_ERR_MSG };
  }
  // return {
  //   statusCode: axiosError.response?.status
  //     ? axiosError.response?.data?.statusCode
  //     : axiosError.code,
  //   message: axiosError.response?.data?.message
  //     ? axiosError.response?.data?.message
  //     : INTERNAL_ERR_MSG,
  // };
};

const apolloErrorBuilder = (apolloError: ApolloError): IAppError => {
  if (apolloError.networkError)
    return { statusCode: GLOBAL_STATUS_CODE, message: NEWTWORK_ERR_MSG };
  if (apolloError.graphQLErrors)
    return graphqlErrorBuilder(apolloError.graphQLErrors[0]);
  return { statusCode: GLOBAL_STATUS_CODE, message: INTERNAL_ERR_MSG };
};

const graphqlErrorBuilder = (graphQLError: GraphQLError): IAppError => {
  const statusCode = graphQLError.extensions.codeNum as number;
  switch (graphQLError.extensions.code) {
    case "UNAUTHENTICATED":
    case "BAD_USER_INPUT":
      return { statusCode, message: graphQLError.message };
    default:
      return { statusCode, message: INTERNAL_ERR_MSG };
  }
};
