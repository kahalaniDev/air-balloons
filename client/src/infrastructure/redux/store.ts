import { configureStore, PreloadedState } from "@reduxjs/toolkit";
import accountReducer, {
  initialState as accountInitialState,
} from "../../features/account/slices/accountSlice";
import balloonsReducer, {
  initialState as balloonInitialState,
} from "../../features/balloons/slices/balloonsSlice";
import { isLoggedIn } from "../../features/account/slices/accountSlice";

type InitialState = {
  account?: typeof accountInitialState;
  balloon?: typeof balloonInitialState;
};

export const createStore = (preloadedState?: PreloadedState<InitialState>) =>
  configureStore({
    preloadedState,
    reducer: { account: accountReducer, balloons: balloonsReducer },
  });

const store = createStore();
store.dispatch(isLoggedIn());

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
