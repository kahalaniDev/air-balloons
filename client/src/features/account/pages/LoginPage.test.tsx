import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { Provider as ReduxProvider } from "react-redux";
import ApolloProvider from "../../../components/apollo/ApolloProvider";
import { createStore } from "../../../infrastructure/redux/store";
import LoginPage from "../pages/LoginPage";

describe("LoginPage", () => {
  beforeAll(() => {});
  test("check if LoginPage match to snapshot", () => {
    expect(
      render(
        <MemoryRouter>
          <ApolloProvider useMocks>
            <ReduxProvider store={createStore()}>
              <LoginPage />
            </ReduxProvider>
          </ApolloProvider>
        </MemoryRouter>
      ).asFragment()
    ).toMatchSnapshot();
  });
});
