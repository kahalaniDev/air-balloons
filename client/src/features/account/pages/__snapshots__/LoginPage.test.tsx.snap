// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage check if LoginPage match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 css-1egif8s-MuiPaper-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Sign in
    </h1>
    <form
      class="MuiBox-root css-164r41r"
      novalidate=""
    >
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="username"
          id="username-label"
        >
          Username
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="username"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="username"
            name="username"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1z7n62"
            >
              <span>
                Username
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          Password
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="current-password"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="password"
            name="password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-173wfxe"
            >
              <span>
                Password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-1koepnq"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonBase-root Mui-disabled  css-17k1s6h-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          Sign In
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
